### PR TITLE
Task 2

### PR DESCRIPTION
**Pull request by elijah-dev**
[Task 2](https://epa.ms/nodejs19-hw2)

### CODE REVIEW

**Comment by FakhriddinAbdurakhimovEPAM** [comment]
Add readme file with tasks description please

File: task-2/src/model/user/model.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  }
+
+  get users() {
+    return this._list;
+  }
+
+  getById(id: string) {
+    const user = this._dict[id];
+
+    if (!user) {
```
Comment (FakhriddinAbdurakhimovEPAM) [inline task-2/src/model/user/model.ts]: I think here should be check for isDeleted == true

File: task-2/src/model/user/model.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    return newUser;
+  }
+
+  updateById(id: string, data: unknown) {
+    const updateUserSchema = getUserSchemaForUpdate();
+    const updateUserData = updateUserSchema.parse(data);
+    const user = this._dict[id] as Record<string, string | number | boolean>;
+
+    if (!user) {
```
Comment (FakhriddinAbdurakhimovEPAM) [inline task-2/src/model/user/model.ts:75]: same check isDeleted

**Review by FakhriddinAbdurakhimovEPAM — CHANGES_REQUESTED** [review]
Looks good, small comments left

File: task-2/src/model/user/model.ts (hunk 1 above)
Comment (elijah-dev) [reply task-2/src/model/user/model.ts]: Yes, I completely forgot about the delete operation. Added

File: task-2/src/model/user/model.ts (hunk 2 above)
Comment (elijah-dev) [reply task-2/src/model/user/model.ts:75]: Delete added

**Review by FakhriddinAbdurakhimovEPAM — APPROVED** [review]

**Pull request merged**